dashboard/lib/widgets/task_grid.dart

```diff
@@ -96,8 +96,80 @@ class TaskGrid extends StatefulWidget {
 
   @override
   State<TaskGrid> createState() => _TaskGridState();
+
+  /// Function to facilitate testing. Do not used this in production. This
+  /// method will sort the supplied CommitStatuses according to the weights
+  /// calculated from the status scores per task and return that sorted list.
+  List<Task> sortedStatuses() {
+    int commitCount = 0;
+    final Map<Task, double> scores = <Task, double>{};
+
+    for (final CommitStatus status in commitStatuses) {
+      commitCount += 1;
+      for (final Task task in status.tasks) {
+        String weightStatus = task.status;
+        if (task.isFlaky || task.isTestFlaky) {
+          // Flaky tasks should be shown after failures and reruns as they take up infra capacity.
+          weightStatus += ' - Flaky';
+        } else if (task.attempts > 1) {
+          // Reruns take up extra infra capacity and should be prioritized.
+          weightStatus += ' - Rerun';
+        }
+
+        // Make the score relative to how long ago it was run.
+        final double score = _statusScores.containsKey(weightStatus)
+            ? _statusScores[weightStatus]! / commitCount
+            : _statusScores['Unknown']! / commitCount;
+
+        scores.update(
+          task,
+          (double value) => value += score,
+          ifAbsent: () => score,
+        );
+      }
+    }
+
+    final List<Task> tasks = scores.keys.toList()
+      ..sort((Task a, Task b) {
+        final int scoreComparison = scores[b]!.compareTo(scores[a]!);
+        if (scoreComparison != 0) {
+          return scoreComparison;
+        }
+        // If the scores are identical, break ties on the name of the task.
+        // We do that because otherwise the sort order isn't stable.
+        if (a.name != b.name) {
+          return a.name.compareTo(b.name);
+        }
+        return a.name.compareTo(b.name);
+      });
+
+    return tasks;
+  }
```

Comment:
Should the code from `_TaskGridState` be removed now?

Reply:
I don't think so since I am only doing parts of that code merely for sorting purposes. I'm not creating the lattice cells and I am not tracking qualified tasks. I suppose I COULD rip it out line for line and simplify the code. Let me look into this.

Reply:
Fixed this to include a sort when the lattice grid view is constructed.